### DISCUSSION TITLE
Frankenphp docker image with xdebug

### DISCUSSION THREAD

**Discussion by guenhter** [php/frankenphp#1034] [Q&A] [answered]
Hi,

is XDebug also working with some of the docker image available under https://hub.docker.com/r/dunglas/frankenphp?

Kind regards

**Comment by dunglas** [comment, marked as answer]
Yes, it works: https://github.com/dunglas/symfony-docker/blob/main/Dockerfile#L60-L63

**Reply by guenhter** [reply]
Thx. Does this then mean images like "dunglas/frankenphp:1.2.3-php8.3-bookworm` are not having xdebug installed?

**Reply by withinboredom** [reply]
That's correct. Why would it be installed by default? There is a huge performance penalty when you install it, not to mention that running it in production would allow anyone on the internet access to your entire system (at least without making some changes to the configuration).

**Reply by nickchomey** [reply]
Extensions can be easily enabled and disabled. Also, you can set this to `trigger`, so that it only works when the request has the right cookie (make it a complicated hash) 

https://xdebug.org/docs/all_settings#start_with_request

Here's how the excellent ddev local dev environment allows for toggling pre-installed xdebug on/off instantly.

This bash script

 https://github.com/ddev/ddev/blob/9c3de4966ba930045e79442299fa2a57e3f9ce4a/pkg/ddevapp/global_dotddev_assets/commands/web/xdebug#L18

Calls this one 

https://github.com/ddev/ddev/blob/9c3de4966ba930045e79442299fa2a57e3f9ce4a/containers/ddev-webserver/ddev-webserver-base-files/usr/local/bin/enable_xdebug#L3

Which uses phpenmod and phpdismod to toggle various extensions (xdebug, xhprof, and others) 


Xdebug definitely shouldnt be enabled by default, but perhaps it could be included and disabled, so that it's there for use in local dev?

**Reply by withinboredom** [reply]
> not to mention that running it in production would allow anyone on the internet access to your entire system **(at least without making some changes to the configuration)**. [emphasis mine]

As you mention, it is relatively straightforward to add if you want it, but I prefer the principle of least surprise, and xdebug being installed (even if disabled), would be surprising. It also further complicates things if you copy over the binaries/libs but not the configuration. This would make some configuration a requirement to use frankenphp, increasing the documentation surface, and I am not sure that is a good idea.
